### PR TITLE
feat: OTC fees

### PR DESCRIPTION
**Pull request by Roznovjak**

### CODE REVIEW

File: pallets/otc/src/lib.rs [hunk 1]
```diff
@@ -291,14 +305,17 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 			let order = <Orders<T>>::get(order_id).ok_or(Error::<T>::OrderNotFound)?;
 
-			Self::execute_order(&order, &who, order.amount_in, order.amount_out)?;
+			let fee = Self::calculate_fee(order.amount_out);
```
Comment (enthusiastmartin) [inline pallets/otc/src/lib.rs:308]: why not to have the fee calculation inside the execute_order ?  you would not need extra parameter and there would be only one place to calculate it.

i get you need to have the fee amount in the event, but the execute order can return the amount.

Comment (Roznovjak) [reply pallets/otc/src/lib.rs:308]: Initially it was there, but I didn't like returning `fee` from `execute_order()`. It doesn't make a lot of sense. But I can change it

Comment (enthusiastmartin) [reply pallets/otc/src/lib.rs:308]: or you can move the deposit event into execute order as well.

Comment (Roznovjak) [reply pallets/otc/src/lib.rs:308]: sounds good, thanks

Comment (Roznovjak) [reply pallets/otc/src/lib.rs:317]: in the latest version, we need fee in some check prior calling execute_order, so I will leave it as it is

**Review by vgantchev — APPROVED** [review]
lgtm

**Review by enthusiastmartin — APPROVED** [review]

**Pull request merged**